apps/learning-snippets/src/test/property-grid/ContextMenuItem.test.tsx

```diff
@@ -0,0 +1,107 @@
+import { PropertyGridComponent } from "@itwin/property-grid-react";
+// __PUBLISH_EXTRACT_END__
+import { queryByText, render, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { buildIModel, insertPhysicalElement, insertPhysicalModelWithPartition, insertSpatialCategory } from "../../utils/IModelUtils";
+import { PropertyGridTestUtils } from "../../utils/PropertyGridTestUtils";
+
+describe("Property grid", () => {
+  describe("Learning snippets", () => {
+    describe("Context menu item", () => {
+      const user = userEvent.setup();
+
+      before(async function () {
+        await initializePresentationTesting({
+          backendProps: {
+            caching: {
+              hierarchies: {
+                mode: HierarchyCacheMode.Memory,
+              },
+            },
+          },
+          testOutputDir: join(__dirname, "output"),
+          backendHostProps: {
+            cacheDir: join(__dirname, "cache"),
+          },
+          rpcs: [SnapshotIModelRpcInterface, IModelReadRpcInterface, PresentationRpcInterface, ECSchemaRpcInterface],
+        });
+        // eslint-disable-next-line @itwin/no-internal
+        ECSchemaRpcImpl.register();
+      });
+
+      after(async function () {
+        await terminatePresentationTesting();
+      });
+
+      beforeEach(async () => {
+        await PropertyGridTestUtils.initialize();
+      });
+
+      afterEach(async () => {
+        await PropertyGridTestUtils.terminate();
+        sinon.restore();
+        // eslint-disable-next-line @itwin/no-internal
+        UiItemsManager.clearAllProviders();
+      });
+
+      it("Renders context menu item", async function () {
+        const imodel = await buildIModel(this, async (builder) => {
+          const physicalModel = insertPhysicalModelWithPartition({ builder, codeValue: "TestPhysicalModel" });
+          const category = insertSpatialCategory({ builder, codeValue: "Test SpatialCategory" });
+          insertPhysicalElement({ builder, modelId: physicalModel.id, categoryId: category.id });
+          return { category };
+        });
+        sinon.stub(UiFramework, "getIModelConnection").returns(imodel.imodel);
+
+        // __PUBLISH_EXTRACT_START__ PropertyGrid.ExampleContextMenuItem
+        function ExampleContextMenuItem(props: ContextMenuItemProps) {
+          return (
+            // render using `PropertyGridContextMenuItem` to get consistent style
+            <PropertyGridContextMenuItem
+              id="example"
+              title="example"
+              onSelect={async () => {
+                console.log(`Selected property: ${props.record.property.displayLabel}`);
+              }}
+            >
+              Click me!
+            </PropertyGridContextMenuItem>
+          );
+        }
+        // __PUBLISH_EXTRACT_END__
+
+        // __PUBLISH_EXTRACT_START__ PropertyGrid.ExampleContextMenuItemRegister
```

Comment:
I think this should be renamed as it does not Register anything

Reply:
Changed